### PR TITLE
DT-604 fixing bug identified during end to end testing.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
Court hearing comments were not being persisted for a court hearing (event).

### CODE REVIEW

File: src/main/java/net/syscon/elite/service/CourtHearingsService.java [hunk 1]
```diff
@@ -78,6 +78,7 @@ public CourtHearing scheduleHearing(final Long bookingId, final PrisonToCourtHea
                 .offenderBooking(offenderBooking)
                 .offenderCourtCase(courtCase)
                 .startTime(hearing.getCourtHearingDateTime())
+                .commentText(hearing.getComments())
```
Comment (matt-crowson-moj) [inline src/main/java/net/syscon/elite/service/CourtHearingsService.java]: This was the culprit not being persisted.

File: src/test/java/net/syscon/elite/service/CourtHearingsServiceTest.java [hunk 2]
```diff
@@ -112,13 +117,23 @@ void setup() {
...
+        assertThat(courtHearingsService.scheduleHearing(offenderBooking.getBookingId(), PRISON_TO_COURT_HEARING)).isEqualTo(CourtHearing.builder()
                 .id(COURT_EVENT_ID)
-                .date(LocalDate.EPOCH)
+                .date(PRISON_TO_COURT_HEARING.getCourtHearingDateTime().toLocalDate())
                 .location(AgencyTransformer.transform(COURT_LOCATION))
-                .time(LocalDate.EPOCH.atStartOfDay().toLocalTime())
+                .time(PRISON_TO_COURT_HEARING.getCourtHearingDateTime().toLocalTime())
+                .build());
+
+        verify(courtEventRepository).save(CourtEvent.builder()
```
Comment (matt-crowson-moj) [inline src/test/java/net/syscon/elite/service/CourtHearingsServiceTest.java]: Added extra test coverage here.

File: src/main/java/net/syscon/elite/repository/jpa/model/CourtEvent.java [hunk 3]
```diff
@@ -31,7 +31,7 @@
 @Builder
 @NoArgsConstructor
 @AllArgsConstructor
-@EqualsAndHashCode(of = "id", callSuper = false)
+@EqualsAndHashCode(callSuper = false)
```
Comment (Mjwillis) [inline src/main/java/net/syscon/elite/repository/jpa/model/CourtEvent.java]: Removing ID is fine I guess - I assume so you have test equality in your tests but collections with a full set of attributes in equals could cause extra loading of data - not a problem here I suspect

Comment (matt-crowson-moj) [reply src/main/java/net/syscon/elite/repository/jpa/model/CourtEvent.java]: Yeah.  I am really on the fence with this one.  Perhaps we can take this offline.

**Pull request merged**